teuthology/orchestra/remote.py

```diff
@@ -300,6 +300,23 @@ def sudo_write_file(self, path, data, **kwargs):
         """
         self.write_file(path, data, sudo=True, **kwargs)
 
+    def is_mounted(self, path):
+        """
+        Check if the given path is mounted on the remote machine.
+
+        This method checks the contents of "/proc/self/mounts" instead of
+        using "mount" or "findmnt" command since these commands hang when a
+        CephFS client is blocked and its mount point on the remote machine
+        is left unhandled/unmounted.
+
+        :param path: path on remote host
+
+        :raises: :class:`RuntimeError`: unexpected error occurred
+        """
+        self.sh(f'test -d {path}')
```

Comment:
This may hang due to a `stat` on `path` if the mount is broken. I still don't see why this is necessary if you're parsing `/proc/self/mounts`.